src/main/java/org/springframework/samples/petclinic/visit/FakeVisitRepository.java

```diff
@@ -0,0 +1,42 @@
+package org.springframework.samples.petclinic.visit;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import org.springframework.dao.DataAccessException;
+
+public class FakeVisitRepository implements VisitRepository{
+	private HashMap<Integer, Visit> visitMap;
+	
+	public FakeVisitRepository() {
+		visitMap = new HashMap<Integer, Visit>();
+	}
+	
+	public FakeVisitRepository(HashMap<Integer, Visit> visitMap) {
+		this.visitMap = visitMap;
+	}
+	
+	@Override
+	public void save(Visit visit) {
+		visitMap.put(visit.getId(), visit);
+	}
+	
+	public int getVisitId(Visit visit) {
```

Comment:
Is there a reason as to why this function is in the repository?

Reply:
Nope, just wanted it get the Id in a straightforward way. Will use the findByPetId approach instead.